components/team_sidebar/components/team_button.jsx

```diff
@@ -49,10 +49,11 @@ export default class TeamButton extends React.Component {
 
         let btn;
         let content = this.props.content;
+        const shortName = this.props.displayName ? this.props.displayName.substring(0, 2) : '??';
```

Comment:
Looks good to me except for the use of `??`.
@jasonblais Could you please validate `??` if we could use this convention whenever displayName is undefined?

Reply:
Yes this feels weird, if the displayName is not set I would use the name of the team, I don't think `??` will do the job at all

Reply:
@saturninoabril @enahum How do I validate this? Do I need to test the team icons in the sidebar?

Reply:
yes the text displayed in the team icon should never be `??` as that does not tell me what team is it, and apart from that what I'm seeing is that if somehow the `displayName` is not set it will not only set the name as `??` but the tooltip will also be empty

Reply:
@dmeza Which issue is this PR resolving?

I was trying to test the fix on the test server, and it seems to work fine, but I'm not sure if I have the correct repro steps since I'm unable to create a team without a display name.

Reply:
@jasonblais @enahum @saturninoabril the `displayName` passed on the props was null so it displayed a ver ugly error. Since that was not an expected path we set it to `??`. What should I set it to if the `displayName` is null?

Reply:
@dmeza @saturninoabril @enahum how do we handle similar cases elsewhere? E.g. a null channel display name?

Reply:
I really can’t say :/

Reply:
I also don't know how it will happen since that field is being validated and cannot be set to empty.
However, on master, weird thing is that when I changed `Team Name` at Main Menu >> Team Settings >> General from `hey` to `    hey` (with spaces before actual team name), I am getting not uniform team button. But this is different issue. See screenshots.

Team Name: `hey`
![screen shot 2017-11-22 at 7 11 39 pm](https://user-images.githubusercontent.com/5334504/33124890-8c860d4a-cfb9-11e7-8467-a37a8df6331b.png)


Team Name: `    hey`
![screen shot 2017-11-22 at 7 11 17 pm](https://user-images.githubusercontent.com/5334504/33124861-6ad894f6-cfb9-11e7-8424-84e6401734f8.png)

Reply:
@saturninoabril @dmeza Do we do this kind of validation for the channel name?

Reply:
Yes, there's minimum channel length of 2, both implemented at server and webapp.

Reply:
@saturninoabril Sorry I meant how do we handle a null channel display name?

Reply:
I don't think we are handling that since we guarantee that the display_name cannot be null.

Reply:
Can we do the same for the team display name? 

We should be consistent how we handle null channel and team display names? Unless of course there's a good reason to deviate.

Reply:
Honestly, I have not seen that happened to team display name (being null). Maybe, the first thing is to request from @dmeza about the repro steps on how he observed null `displayName` being passed to `TeamButton`.

Reply:
I know this seems odd, we thought so at the time, but it is possible. By looking at code we could not figure out why, debugging was hard. It completely killed production at one point during launch. So it's best we just guard against this knowing it's not a likely case, but is possible.

Again, we didn't do the ?? for style, we did it because the web app would not load without fixing this.

Reply:
@dmeza Waiting on design team's feedback -- just curious if it's easy to set a non-empty tooltip for this case like `Name undefined`? Just want to gauge the level of effort,

Reply:
@dmeza The `??` should be okay in the team icon. I'm not sure what else would be better instead. It would be nice if there was a tooltip in that case with text `Name undefined`, but depends on level of effort for adding it.